server/src/main/java/org/candlepin/dto/CandlepinDTO.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2009 - 2017 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.dto;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import java.io.Serializable;
+
+
+
+/**
+ * The CandlepinDTO is the base class for all DTOs, which provides common data and functionality.
+ *
+ * @param <T>
+ *  DTO type extending this class; should be the name of the subclass
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public abstract class CandlepinDTO<T extends CandlepinDTO> implements Cloneable, Serializable {
```

Comment:
I'm confused why `CandlepinDTO` uses a generic type when it doesn't seem to have any generic methods.  I.e. method returning something of type `T`